### PR TITLE
CLDC-3570 Validations all other shared ownership types

### PR DESCRIPTION
**Pull request by Dinssa**
This branch changes the error message given when a compound validation fails as a result of values in questions Q89, Q90, Q92, Q96. And, updates the partial guidance.

Screenshot 2024-08-06 at 21 28 52

### CODE REVIEW

File: app/views/form/_numeric_question.html.erb [hunk 1]
```diff
@@ -14,7 +14,7 @@
   suffix_text: question.suffix_label(@log),
   value: format_money_input(log: @log, question:),
   inputmode: "numeric",
-  pattern: "\d*\.?\d*",
+  pattern: "\\d*\\.?\\d*",
```
Comment (Dinssa) [inline app/views/form/_numeric_question.html.erb:17]: Update regex pattern to escape backslashes

File: app/views/form/guidance/_financial_calculations_shared_ownership.html.erb [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+<% value_page = log.form.get_question("value", log).page %>
 <%= govuk_details(summary_text: "How the financial values are calculated") do %>
   <p class="govuk-body">
-  The mortgage amount (<%= govuk_link_to "Q92", send("#{log.class.name.underscore}_#{log.form.get_question('mortgage', log).page.id}_path", log) %>),
-   cash deposit (<%= govuk_link_to "Q95", send("#{log.class.name.underscore}_#{log.form.get_question('deposit', log).page.id}_path", log) %>)
-   and cash discount (<%= govuk_link_to "Q97", send("#{log.class.name.underscore}_#{log.form.get_question('cashdis', log).page.id}_path", log) %>)
-   added together must equal
-   the purchase price (<%= govuk_link_to "Q88", send("#{log.class.name.underscore}_#{log.form.get_question('value', log).page.id}_path", log) %>)
-   multiplied by the percentage equity stake (<%= govuk_link_to "Q89", send("#{log.class.name.underscore}_#{log.form.get_question('equity', log).page.id}_path", log) %>)
+    <% if log.type == 18 %>
```
Comment (Dinssa) [inline app/views/form/guidance/_financial_calculations_shared_ownership.html.erb:3]: This condition shows one guidance text for Social HomeBuy (log type: 18) with the cash discount and another guidance text for all other shared ownership types without the cash discount.

File: app/views/form/guidance/_financial_calculations_shared_ownership.html.erb [hunk 3]
```diff
@@ -1,28 +1,20 @@
...
 <%= govuk_details(summary_text: "How the financial values are calculated") do %>
   <p class="govuk-body">
     <% if log.type == 18 %>
       The mortgage amount <%= question_link("mortgage", log, current_user) %>,
       cash deposit <%= question_link("deposit", log, current_user) %>,
       and cash discount <%= question_link("cashdis", log, current_user) %>
-      added together must equal
-      the purchase price <%= question_link("value", log, current_user) %>
-      <% stairbought_page = log.form.get_question("stairbought", log).page %>
-      <% equity_page = log.form.get_question("equity", log).page %>
```
Comment (Dinssa) [inline app/views/form/guidance/_financial_calculations_shared_ownership.html.erb]: This variable wasn't used so I removed it

File: app/views/form/guidance/_financial_calculations_shared_ownership.html.erb (hunk 2 above)
Comment (Dinssa) [reply app/views/form/guidance/_financial_calculations_shared_ownership.html.erb:3]: Conditional now just used for the first part of the guidance text

File: app/models/validations/sales/sale_information_validations.rb [hunk 4]
```diff
@@ -296,15 +296,13 @@ def check_staircasing_non_socialhomebuy_mortgage(record)
                                           deposit: record.field_formatted_as_currency("deposit"),
                                           mortgage_and_deposit_total: record.field_formatted_as_currency("mortgage_and_deposit_total"),
                                           value: record.field_formatted_as_currency("value"),
-                                          equity: "#{record.equity}%",
```
Comment (Dinssa) [inline app/models/validations/sales/sale_information_validations.rb]: Unused

**Review by kosiakkatrina — APPROVED** [review]

**Pull request merged**